File: src/main/java/com/cfgglobal/test/config/app/ApplicationProperties.java
package com.cfgglobal.test.config.app;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.util.List;

@Data
@ConfigurationProperties("app")
public class ApplicationProperties {

    public static String myUserClass;

    public static String enumPackage;

    public static List<String> enums;

    public static String[] entityScanPackage;


    String domain;

    String rateDataPath;

    String userClass;

    String userCookie;


    Jwt jwt;

    DateSettings dateSettings;

    @Data
    public static class Jwt {
        String header = "Authorization";
        Long expiresIn = 864000L;
        String secret = "queenvictoria";
        String cookie = "AUTH-TOKEN";
        String param = "token";
        String anonymousUrls;
    }

    @Data
    public static class DateSettings {
        String dateType = "ZonedDateTime";
    }
}

File: src/main/kotlin/com/github/leon/aci/security/ApplicationProperties.kt
package com.github.leon.aci.security

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties("app")
class ApplicationProperties {


    var domain: String? = null

    var rateDataPath: String? = null

    var userClass: String? = null

    var userCookie: String? = null


    var jwt: Jwt? = null


    class Jwt {
        internal var header = "Authorization"
        internal var expiresIn: Long? = 864000L
        internal var secret = "queenvictoria"
        internal var cookie = "AUTH-TOKEN"
        internal var param = "token"
        internal var anonymousUrls: String? = null
    }

    companion object {

        var myUserClass: String? = null

        var enumPackages: List<String> = listOf()

        var enums: List<String> = listOf()

        var entityScanPackage: List<String> = listOf()

        var dateType = "ZonedDateTime"

        var displayTimeZone = "Pacific/Auckland"

        var dbTimeZone = "UTC"
    }
}
